kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/test/java/org/kie/server/services/optaplanner/SolverServiceBaseTest.java

```diff
@@ -0,0 +1,220 @@
+        assertBuildFactoryWasInvoked(null);
+        assertSuccess(response, String.format(SOLVER_CREATED_SUCCESSFULLY_MESSAGE, SOLVER_ID, CONTAINER_ID));
+    }
+
+    @Test
+    public void createSolverWithGlobalScanSetSuccessful() {
+        solverInstance.setSolverConfigFile(SOLVER_CONFIG_GLOBAL_SCAN_SET);
+        doReturn(solver).when(serviceBase).newSolver(any(SolverFactory.class));
+
+        ServiceResponse<SolverInstance> response = serviceBase.createSolver(CONTAINER_ID, SOLVER_ID, solverInstance);
+        assertBuildFactoryWasInvoked(Arrays.asList(TestConfigProvider1.PACKAGE1,
+                                                   TestConfigProvider1.PACKAGE2,
+                                                   TestConfigProvider1.PACKAGE3,
+                                                   TestConfigProvider2.PACKAGE1));
+        assertSuccess(response, String.format(SOLVER_CREATED_SUCCESSFULLY_MESSAGE, SOLVER_ID, CONTAINER_ID));
+    }
+
+    @Test
+    public void createSolverWithFilteredScanSetSuccessful() {
+        solverInstance.setSolverConfigFile(SOLVER_CONFIG_FILTERED_SCAN_SET);
+        doReturn(solver).when(serviceBase).newSolver(any(SolverFactory.class));
+
+        ServiceResponse<SolverInstance> response = serviceBase.createSolver(CONTAINER_ID, SOLVER_ID, solverInstance);
+        assertBuildFactoryWasInvoked(Collections.singletonList(PRE_CONFIGURED_PACKAGE_INCLUDE),
+                                     Arrays.asList(TestConfigProvider1.PACKAGE1,
+                                                   TestConfigProvider1.PACKAGE2,
+                                                   TestConfigProvider1.PACKAGE3,
+                                                   TestConfigProvider2.PACKAGE1,
+                                                   PRE_CONFIGURED_PACKAGE_EXCLUDE));
+        assertSuccess(response, String.format(SOLVER_CREATED_SUCCESSFULLY_MESSAGE, SOLVER_ID, CONTAINER_ID));
+    }
+
+    private void assertFailure(ServiceResponse<SolverInstance> response, String messagePrefix) {
+        assertThat(response.getType()).isEqualTo(ServiceResponse.ResponseType.FAILURE);
+        assertThat(response.getMsg()).startsWith(messagePrefix);
+    }
+
+    private void assertSuccess(ServiceResponse<SolverInstance> response, String messagePrefix) {
+        assertThat(response.getType()).isEqualTo(ServiceResponse.ResponseType.SUCCESS);
+        assertThat(response.getMsg()).startsWith(messagePrefix);
+    }
+
+    private void assertBuildFactoryWasInvoked(List<String> expectedExcludedPackages) {
+        assertBuildFactoryWasInvoked(null, expectedExcludedPackages);
+    }
+
+    private void assertBuildFactoryWasInvoked(List<String> expectedIncludedPackages, List<String> expectedExcludedPackages) {
+        verify(serviceBase).newSolver(solverFactory.capture());
+        if (expectedExcludedPackages != null) {
+            assertThat(solverFactory.getValue().getSolverConfig().getScanAnnotatedClassesConfig()).isNotNull();
```

Comment:
Soft suggestion: assigning `solverFactory.getValue().getSolverConfig().getScanAnnotatedClassesConfig()` to a local variable will make the following lines more readable